src/query/query_eval.rs

```diff
@@ -33,34 +34,87 @@ fn eval_latest<'a>(
 fn eval_lookup<'a>(
     index: &'a Index,
     lookup_field: LookupLhs,
-    value: &str,
+    value: LookupRhs,
 ) -> Result<Vec<&'a Packet>, QueryError> {
     Ok(index
         .packets
         .iter()
         .filter(|packet| match lookup_field {
-            LookupLhs::Id => packet.id == value,
-            LookupLhs::Name => packet.name == value,
-            LookupLhs::Parameter(param_name) =>
-                packet.parameter_equals(param_name, ParameterValue::String(value))
+            LookupLhs::Id => match value {
+                LookupRhs::String(str) => packet.id == str,
+                _ => false
+            },
+            LookupLhs::Name => match value {
+                LookupRhs::String(str) => packet.name == str,
+                _ => false
+            },
+            LookupLhs::Parameter(param_name) => packet.parameter_equals(param_name, &value)
         })
         .collect())
 }
 
+impl Packet {
+    /// Check if a packet parameter is equal to a test value
+    ///
+    /// This will get the parameter `param_name` from the current Packet and then test for
+    /// equality of the test value `value`.
+    ///
+    /// # Arguments
+    /// * `param_name` - A string slice that holds the name of the parameter to test
+    /// * `value` - A LookupRhs which holds the value to check equality for, can be a boolean, a
+    ///             string or a number
+    ///
+    /// # Return
+    /// * bool - true if the current packet has a parameter called `param_name` and its value
+    ///          is equal to the LookupRhs.
+    fn parameter_equals(&self, param_name: &str, value: &LookupRhs) -> bool {
+        if let Some(json_value) = self.get_parameter(param_name) {
```

Comment:
I stared at this for a bit wondering where `json_value` came from before realising this is a destructuring of the result of `self.get_parameter`!